src/MatrixFields/invertible_field_matrix.jl

```diff
@@ -0,0 +1,48 @@
+"""
+    InvertibleFieldMatrix(A, b, [alg])
+
+A wrapper that combines a `FieldMatrix` `A` with a `FieldMatrixSolver` that can
+be used to solve the equation `A * x = b` for `x`, where `x` and `b` are both
+`FieldVector`s. Similar to a `LinearAlgebra.Factorization`, this wrapper can be
+passed to `ldiv!`, whereas a regular `FieldMatrix` cannot be passed to `ldiv!`.
+
+By default, the `FieldMatrixSolverAlgorithm` `alg` is set to a
+[`BlockDiagonalSolve`](@ref), so a custom `alg` must be specified when `A` is
+not a block diagonal matrix.
+"""
+struct InvertibleFieldMatrix{M, S} <: AbstractDict{FieldNamePair, Any}
```

Comment:
I'm not sure I like this name, the ability to invert a matrix depends on the type of matrix but may also depend on the _values_ in the matrix. So it's not clear that this type will hold up to its name (unless we check upon construction, which sounds expensive).

Is there a different name that we can use that might be more suitable?

Reply:
cc: @dennisYatunin

Reply:
The closest analogy to this in Julia is a `LinearAlgebra.Factorization`, but I don't think we should use a name like that because we're not planning to do any factorization here. Something like `LDivableFieldMatrix` would probably be accurate, but it sounds awful.

We currently do one simple check for invertibility---making sure that there are no missing (zero) blocks along the diagonal. This check only relies on type information, so it's very cheap. We could conceivably add a debug mode to this struct as well, which would compute the full matrix determinant and check that it isn't too close to zero. This would help us catch issues in the Jacobian before they cause the solution to blow up, so we might want it for debugging implicit prognostic EDMF.

Since we already do one check for invertibility and we could potentially add more expensive checks later, maybe `InvertibleFieldMatrix` is fine? I can also modify the docstring to include this information.

Reply:
@dennisYatunin, how about `SquareFieldMatrix`?

Reply:
Or even just `FieldMatrix`?

Reply:
I personally like `SquareFIeldMatrix`, since it's necessary but not sufficient for invertibility

Reply:
Going with `FieldMatrixWithSolver`.